frame/grandpa/src/lib.rs

```diff
@@ -643,10 +658,17 @@ where
 			};
 
 			if res.is_ok() {
-				CurrentSetId::<T>::mutate(|s| {
+				let current_set_id = CurrentSetId::<T>::mutate(|s| {
 					*s += 1;
 					*s
-				})
+				});
+
+				let max_set_id_session_entries = (T::MaxSetIdSessionEntries::get() as u64).max(1);
```

Comment:
```suggestion
				let max_set_id_session_entries = T::MaxSetIdSessionEntries::get().max(1);
```

Reply:
what does `max(1)` do? sets a default max to 1?

Reply:
Yeah, the way the code is written there will always be one entry in the map regardless if you set this value to 0. I could change the code to respect the 0 limit, but didn't think it was worth the extra complexity, in particular because this will always be `> 0` for any practical implementations (unless someone is using this without staking and/or without equivocation handling).